### PR TITLE
refactor(core): Request mapper consolidation

### PR DESCRIPTION
**Pull request by kpavlov**
### New Features

  * Added broader support for MCP request types, including task operations, cancellation, logging levels, completions, and resource handling.
  * Added support for reading declared client extensions and related metadata.
  * Improved compatibility across MCP protocol versions, including legacy request formats and version-specific payloads.

### Bug Fixes

  * Invalid or non-object request parameters now produce clear request-mapping errors instead of being silently accepted.
  * Improved handling of task status defaults and numeric cancellation request IDs.

### CODE REVIEW

**Comment by kpavlov** [comment]
@coderabbitai review

File: .agents/skills/tachyon-development/SKILL.md [hunk 1]
```diff
@@ -1,68 +1,73 @@
...
-- Fall back to raw `JsonUnit` + AssertJ (`assertThatJson(actual).isEqualTo(expected)`, full `// language=JSON` block, dynamic values via `.formatted(...)`) where `JsonRpcResponseAssert` doesn't model the shape — nested payloads like `tasks/get`'s `result.result.*`, or clients/modules outside tachyon-testkit (official `io.modelcontextprotocol.sdk`, Kotlin's stack, etc.). Same full-payload rule applies: no `inPath(...)` fragments or field-by-field getters — both let an extra, missing, or wrong-shaped field pass silently. Derive `expected` from a real run (fail once, read the reported value, verify it's correct, lock it in) — never a hand-typed guess. `whenIgnoringPaths`/`IGNORING_ARRAY_ORDER`/`IGNORING_EXTRA_FIELDS`/`TREATING_NULL_AS_ABSENT` only for genuinely non-deterministic parts (timestamps, generated ids) — never the default style.
+- Prefer E2E; use unit tests only for edge cases E2E cannot cover.
+- Test intent, not line coverage. No tautologies or abstract-base-only tests; exercise shared behavior
+  through concrete implementations or E2E.
+- Java: JUnit 6 + Pioneer, AssertJ fluent, short spec-reference comments, parameterized tests when appropriate.
```
[the remark this answers is not in the document]
Comment (kpavlov) [reply .agents/skills/tachyon-development/SKILL.md:38]: not important
